lib/url-state-machine.js

```diff
@@ -575,6 +575,16 @@ URLStateMachine.prototype["parse scheme start"] = function parseSchemeStart(c, c
 URLStateMachine.prototype["parse scheme"] = function parseScheme(c, cStr) {
   if (infra.isASCIIAlphanumeric(c) || c === p("+") || c === p("-") || c === p(".")) {
     this.buffer += cStr.toLowerCase();
+  } else if (c === p(":") && this.buffer.length === 1 &&
+             infra.isASCIIAlpha(this.buffer.codePointAt(0)) &&
+             this.input[this.pointer + 1] === p("\\") &&
+             !this.stateOverride) {
+    this.url.scheme = "file";
+    this.url.host = "";
+    // Preserve original case of drive letter from input
+    const driveLetter = String.fromCodePoint(this.input[this.pointer - 1]);
+    this.buffer = `${driveLetter}:`;
```

Comment:
The spec does `this.buffer = this.buffer + ":"`, it does not do this driveLetter thing.